services/scoop/scoop-version.service.js

```diff
@@ -0,0 +1,95 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { NotFound } = require('..')
+const {
+  ConditionalGithubAuthV3Service,
+} = require('../github/github-auth-service')
+const { fetchJsonFromRepo } = require('../github/github-common-fetch')
+const { renderVersionBadge } = require('../version')
+
+const gitHubRepoRegExp = new RegExp(
+  'https://github.com/(?<user>.*)/(?<repo>.*)/?'
```

Comment:
Note this works well without the trailing slash, but if there is trailing slash you will get:

```js
gitHubRepoRegExp.exec('https://github.com/lukesampson/scoop/')
// user: "lukesampson/scoop"
// repo: ""
```

```suggestion
  'https://github.com/(?<user>.*?)/(?<repo>.*?)(/|$)'
```

Reply:
Thanks for pointing this out @kidonng. Would you be interested in helping us finish this implementation?

Reply:
Did you mean finishing this PR? I see it's basically working but lacks the ability to cache `buckets.json`, I'd like to help but I can't find documentation about handling caching.

Reply:
Yes, potentially finishing the PR if you're up for it!

Caching the `bucket.json` doesn't have to be very fancy. On the first request, get the file and store its contents in a class field that is reused for all subsequent requests. :)